pywr-schema/src/nodes/river.rs

```diff
@@ -193,7 +384,84 @@ impl RiverNode {
+                            sub_name: Self::output_sub_name().map(String::from),
+                        })?;
+                    let metric = MetricF64::NodeInFlow(inflow_idx);
+
+                    let p = pywr_core::parameters::DelayParameter::new(
+                        name,
+                        metric,
+                        *delay,
+                        initial_value.load(args.tables)?,
+                    );
+                    let delay_idx = network.add_parameter(Box::new(p))?;
+
+                    // Apply it as a constraint on the input node.
+                    let metric: MetricF64 = delay_idx.into();
+                    network.set_node_max_flow(
+                        self.meta.name.as_str(),
+                        Self::input_sub_name(),
+                        metric.clone().into(),
+                    )?;
+                    network.set_node_min_flow(self.meta.name.as_str(), Self::input_sub_name(), metric.into())?;
+                }
+                RoutingMethod::Muskingum {
+                    travel_time,
+                    weight,
+                    initial_condition,
+                } => {
+                    // Create the delay parameter using the node's name as the parent identifier
+                    let name = ParameterName::new("muskingum", Some(self.meta.name.as_str()));
+                    let inflow_idx = network
+                        .get_node_index_by_name(self.meta.name.as_str(), Self::output_sub_name())
+                        .ok_or_else(|| SchemaError::CoreNodeNotFound {
+                            name: self.meta.name.clone(),
+                            sub_name: Self::output_sub_name().map(String::from),
+                        })?;
+                    let outflow_idx = network
+                        .get_node_index_by_name(self.meta.name.as_str(), Self::input_sub_name())
+                        .ok_or_else(|| SchemaError::CoreNodeNotFound {
+                            name: self.meta.name.clone(),
+                            sub_name: Self::input_sub_name().map(String::from),
+                        })?;
+
+                    let travel_time = travel_time.load(network, args, Some(&self.meta.name))?;
+                    let weight = weight.load(network, args, Some(&self.meta.name))?;
+
+                    let muskingum_parameter = MuskingumParameter::new(
+                        name.clone(),
+                        MetricF64::NodeInFlow(inflow_idx),
+                        MetricF64::NodeOutFlow(outflow_idx),
+                        travel_time,
+                        weight,
+                        initial_condition.clone().into(),
+                    );
+
+                    let muskingum_idx = network.add_multi_value_parameter(Box::new(muskingum_parameter))?;
+
+                    let factors = Relationship::new_ratio_factors(
+                        &[(muskingum_idx.clone(), "inflow_factor".to_string()).into(), 1.0.into()],
```

Comment:
Is the order of factors here correct? When the add node is created the inflow node is second in the node list.